hdfs/src/object_store/hdfs.rs

```diff
@@ -111,6 +112,58 @@ impl Display for HadoopFileSystem {
     }
 }
 
+struct HdfsMultiPartUpload {
+    location: Path,
+    hdfs: Arc<HdfsFs>,
+    content: Arc<Mutex<HashMap<usize, Vec<u8>>>>,
```

Comment:
This may lead to too much memory pressure, since it has to keep all of the contents in memory before sending to the HDFS.